File: Java/Bit_Manipulation/_08_PowOfTwo.java
// 08. Check if a given number is in power of 2 or not.
public class _08_PowOfTwo {
    private static int isPowerOfTwo(int num) {
        return (num & ~(num & (num - 1)));
    }

    public static void main(String[] args) {
        int num = 255;
        if (isPowerOfTwo(num) != 0)
            System.out.println(num + " is in power of 2 ");
        else
            System.out.println(num + " is not in power of 2 ");
    }
}

File: C++/Bit_Manipulation/_08_PowOfTwo.cpp
// 08. Check if a given number is in power of 2 or not.
#include <iostream>
using namespace std;
int isPowerOfTwo(int num)
{
    return (num & ~(num & (num - 1)));
}

int main()
{
    int num = 255;
    if (isPowerOfTwo(num) != 0)
        cout << num << " is in power of 2 " << endl;
    else
        cout << num << " is not in power of 2 " << endl;
}